### PR TITLE
drivers/serial: stm32: Simplify code dependency checks

### PR DESCRIPTION
**Pull request by erwango**
Rework stm32 serial driver to simplify code activation depending on series.
Since more and more series are present in Zephyr, prefer use of definitions from STM32Cube packages rather than series Kconfig symbols which will require long and heavy `#ifdef` lines

### CODE REVIEW

**Comment by erwango** [comment]
@zhaozhongchen , @pic16f887, FYI

**Review by ydamigos — APPROVED** [review]
LGTM

File: drivers/serial/uart_stm32.c [hunk 1]
```diff
@@ -331,12 +331,17 @@ static int uart_stm32_init(struct device *dev)
...
-	while ((!(LL_USART_IsActiveFlag_TEACK(UartInstance))) ||
-	      (!(LL_USART_IsActiveFlag_REACK(UartInstance))))
+#ifdef USART_ISR_TEACK
+	/* Wait until TEACK flag is set */
+	while (!(LL_USART_IsActiveFlag_TEACK(UartInstance)))
+		;
+#endif /* !USART_ISR_TEACK */
+
+#ifdef USART_ISR_REACK
+	/* Wait until TEACK flag is set */
```
Comment (lpereira) [inline drivers/serial/uart_stm32.c]: This is just the comment, but shouldn't this be the REACK flag?

Comment (erwango) [reply drivers/serial/uart_stm32.c]: Indeed, thanks!

**Review by nashif — APPROVED** [review]

**Pull request merged**